### PR TITLE
Add json field

### PR DESCRIPTION
**Pull request by samfishman**
**Jira:** https://clever.atlassian.net/browse/DIS2-226

**Overview:**
Starts storing JSON payloads in a new `json_workload` column in postgres to allow faster searching across jobs. Also exposes a new filter (called `workload_contains`) on the `get_jobs` db function to actually make these searches. This allows searches such as `/api/jobs?workload_contains={"district_id":"abc123"}`. NOTE: there is no UI for this yet, but the workload_contains query param is passed through from the angular frontend to the api endpoint, so you can URLs such as `/jobs/?workload_contains=%7B%22msg%22:%22hi%22%7D` (that's `{"msg":"hi"}` url-encoded) work and render the matching jobs in the frontend.

**Testing:**
Tests were added to make sure that the new filtering works as expected in test/db.coffee.

### CODE REVIEW

File: lib/postgres.coffee [hunk 1]
```diff
@@ -45,7 +55,7 @@ module.exports = class Postgres extends GearmanDatabase
 
   add_job: (job, cb) =>
     @_get_postgres cb, (db, cb_pg) =>
-      db.query 'INSERT INTO jobs VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)', @_job_to_array(job), return_only_err(cb_pg)
+      db.query 'INSERT INTO jobs VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)', @_job_to_array(job), return_only_err(cb_pg)
```
Comment (natebrennand) [inline lib/postgres.coffee]: might be time to make this insert statement a little more explicit:

like:
```
INSERT INTO films (code, title, did, date_prod, kind)
    VALUES ('T_601', 'Yojimbo', 106, '1961-06-16', 'Drama');
```
- https://www.postgresql.org/docs/current/static/sql-insert.html

File: lib/postgres.coffee [hunk 2]
```diff
@@ -75,6 +85,10 @@ module.exports = class Postgres extends GearmanDatabase
       base_query += ("#{key} LIKE '%' || $#{params.length} || '%'" for key in ["workload", "data", "warnings", "handle", "function"]).join(" OR ")
       base_query += ")"
 
+    if query.workload_contains?
```
Comment (natebrennand) [inline lib/postgres.coffee]: should this use `is_json` to filter out invalid `workload_contains` queries?

File: lib/sqlite.coffee [hunk 3]
```diff
@@ -53,6 +53,40 @@ module.exports = class SQLite extends GearmanDatabase
...
+        contains_obj = JSON.parse query.workload_contains
+      catch err
+        return cb err
+
+      # sqlite doesn't support a JSON contains operator like postgresql, so we
+      # need to simulate the behavior. `json_extract(data, path)` accesses the
+      # value stored in `data` found by following the JSON path specified by
+      # `path`, where `$` denotes the root object. So, the path '$.x' would
+      # yield 5 in '{"x":5}'.
+      # We perform a DFS on the `workload_contains` object, generating a new
```
Comment (natebrennand) [inline lib/sqlite.coffee]: Is doing a DFS strictly necessary? Almost all of our payloads use a single layer JSON payload

**Comment by natebrennand** [comment]
well, I guess if the nested JSON is already tested and works, it's awesome. I hope that scope creep didn't take too long 😺 

Only real question is on the filtering out invalid `workload_contains` queries on Postgres.

----

LGTM

File: lib/postgres.coffee (hunk 2 above)
Comment (samfishman) [inline lib/postgres.coffee]: Good idea. 1b35de1

File: lib/postgres.coffee (hunk 1 above)
Comment (samfishman) [inline lib/postgres.coffee]: 6afc291

**Pull request merged**